### PR TITLE
Allow building on *BSD systems

### PR DESCRIPTION
**Pull request by voidpin**
I'm currently packaging LXQt for NetBSD and there are a few patches needed.
Please allow building on *BSD systems. Thanks!

### CODE REVIEW

File: sudo.cpp [hunk 1]
```diff
@@ -39,7 +39,13 @@
 #include <QProcessEnvironment>
 #include <QTimer>
 #include <QRegularExpression>
+#if defined(__Linux__)
```
Comment (palinek) [inline sudo.cpp]: Would you mind using the "Qt style" OS detection?
```c++
/*
   The operating system, must be one of: (Q_OS_x)

     DARWIN   - Any Darwin system (macOS, iOS, watchOS, tvOS)
     MACOS    - macOS
     IOS      - iOS
     WATCHOS  - watchOS
     TVOS     - tvOS
     WIN32    - Win32 (Windows 2000/XP/Vista/7 and Windows Server 2003/2008)
     WINRT    - WinRT (Windows Runtime)
     CYGWIN   - Cygwin
     SOLARIS  - Sun Solaris
     HPUX     - HP-UX
     LINUX    - Linux [has variants]
     FREEBSD  - FreeBSD [has variants]
     NETBSD   - NetBSD
     OPENBSD  - OpenBSD
     INTERIX  - Interix
     AIX      - AIX
     HURD     - GNU Hurd
     QNX      - QNX [has variants]
     QNX6     - QNX RTP 6.1
     LYNX     - LynxOS
     BSD4     - Any BSD 4.4 system
     UNIX     - Any UNIX BSD/SYSV system
     ANDROID  - Android platform
     HAIKU    - Haiku

   The following operating systems have variants:
     LINUX    - both Q_OS_LINUX and Q_OS_ANDROID are defined when building for Android
              - only Q_OS_LINUX is defined if building for other Linux systems
     FREEBSD  - Q_OS_FREEBSD is defined only when building for FreeBSD with a BSD userland
              - Q_OS_FREEBSD_KERNEL is always defined on FreeBSD, even if the userland is from GNU
*/

```

**Comment by voidpin** [comment]
No problem. Is this ok now?

File: sudo.cpp [hunk 2]
```diff
@@ -39,7 +39,13 @@
 #include <QProcessEnvironment>
 #include <QTimer>
 #include <QRegularExpression>
+#if defined(__Q_OS_LINUX__)
```
Comment (palinek) [inline sudo.cpp]: Did you test it? AFAIK it should be `Q_OS_LINUX`.

**Comment by voidpin** [comment]
Give me an hour or so. I'm setting up the build right now.

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Pull request closed without merging**